sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/model/ModelBridgeInternal.java

```diff
@@ -0,0 +1,405 @@
+    }
+
+    public static <T> ConcurrentMap<String, QueryMetrics> queryMetricsMap(FeedResponse<T> feedResponse) {
+        return feedResponse.queryMetricsMap();
+    }
+
+    public static <T> ConcurrentMap<String, QueryMetrics> queryMetrics(FeedResponse<T> feedResponse) {
+        return feedResponse.queryMetrics();
+    }
+
+    public static String toLower(RequestVerb verb) {
+        return verb.toLowerCase();
+    }
+
+    public static boolean isV2(PartitionKeyDefinition pkd) {
+        return pkd.getVersion() != null && PartitionKeyDefinitionVersion.V2.val == pkd.getVersion().val;
+    }
+
+    public static PartitionKeyInternal getNonePartitionKey(PartitionKeyDefinition partitionKeyDefinition) {
+        return partitionKeyDefinition.getNonePartitionKeyValue();
+    }
+
+    public static PartitionKeyInternal getPartitionKeyInternal(PartitionKey partitionKey) {
+        return partitionKey.getInternalPartitionKey();
+    }
+
+    public static PartitionKey partitionKeyfromJsonString(String jsonString) {
+        return PartitionKey.fromJsonString(jsonString);
+    }
+
+    public static Object getPartitionKeyObject(PartitionKey right) {
+        return right.getKeyObject();
+    }
+
+    public static String getAltLink(Resource resource) {
+        return resource.getAltLink();
+    }
+
+    public static void setAltLink(Resource resource, String altLink) {
+        resource.setAltLink(altLink);
+    }
+
+    public static void setResourceSelfLink(Resource resource, String selfLink) {
+        resource.setSelfLink(selfLink);
+    }
+
+    public static void setTimestamp(Resource resource, OffsetDateTime date) {
+        resource.setTimestamp(date);
+    }
+
+    public static void validateResource(Resource resource) {
+        Resource.validateResource(resource);
+    }
+
+    public static <T> void setProperty(JsonSerializable jsonSerializable, String propertyName, T value) {
+        jsonSerializable.set(propertyName, value);
+    }
+
+    public static ObjectNode getObject(JsonSerializable jsonSerializable, String propertyName) {
+        return jsonSerializable.getObject(propertyName);
+    }
+
+    public static void remove(JsonSerializable jsonSerializable, String propertyName) {
+        jsonSerializable.remove(propertyName);
+    }
+
+    public static Object getValue(JsonNode value) {
+        return JsonSerializable.getValue(value);
+    }
+
+    public static CosmosError createCosmosError(ObjectNode objectNode) {
+        return new CosmosError(objectNode);
+    }
+
+    public static CosmosError createCosmosError(String jsonString) {
+        return new CosmosError(jsonString);
+    }
+
+    public static void populatePropertyBagJsonSerializable(JsonSerializable jsonSerializable) {
+        jsonSerializable.populatePropertyBag();
+    }
+
+    public static void setMapper(JsonSerializable jsonSerializable, ObjectMapper om) {
+        jsonSerializable.setMapper(om);
```

Comment:
why do we need the setMapper?

Reply:
setMapper is used in a constructor in Document.java class. 

```
/**
     * Initialize a document object from json string.
     *
     * @param jsonString the json string that represents the document object.
     * @param objectMapper the custom object mapper
     */
    Document(String jsonString, ObjectMapper objectMapper) {
        // TODO: Made package private due to #153. #171 adding custom serialization options back.
        super(jsonString);
        setMapper(this, objectMapper);
    }
```

Reply:
I checked again, and this constructor is not used at all, I will see if we can remove this constructor and hence setMapper can be removed as well.